dom.go

```diff
@@ -173,7 +203,7 @@ func (e *Element) Reconcile(oldComp Component) {
 			replaceNode(newChild.Node(), oldChild.Node())
 		}
 		for i := len(e.Children); i < len(oldElement.Children); i++ {
-			removeNode(oldElement.Children[i].Node())
+			oldElement.Children[i].Unmount()
```

Comment:
In this function there is `appendChild` and `replaceNode`, where it is the parent's responsibility to manage the proper insertion of the child's DOM node into the document. It is very inconsistent to not have `removeNode` here as well. We can still have some method to inform the child that it is about to be unmounted, but the actual `removeNode` should not be done by the child.